giraffe/src/components/AnnotationHoverLayer.tsx

```diff
@@ -0,0 +1,42 @@
+import React, {FunctionComponent} from 'react'
+import {AnnotationMark} from '../types'
+import {AnnotationLayerProps} from './AnnotationLayer'
+import {AnnotationTooltip} from './AnnotationTooltip'
+
+interface AnnotationHoverLayerProps extends AnnotationLayerProps {
+  annotationPositions: AnnotationMark[]
+  boundingReference: DOMRect
+  hoverRowIndices: number[]
+  width: number
+  height: number
+}
+
+export const AnnotationHoverLayer: FunctionComponent<AnnotationHoverLayerProps> = props => {
+  const {
+    annotationPositions,
+    plotConfig,
+    hoverRowIndices,
+    boundingReference,
+    width,
+    height,
+  } = props
+
+  const annotationTooltipData = annotationPositions.filter((_, i) =>
+    hoverRowIndices.includes(i)
+  )
+
+  return (
+    <>
+      {annotationTooltipData.map(annotationData => (
+        <AnnotationTooltip
+          key={`annotation-tooltip-${annotationData.dimension}-${annotationData.startValue}-${annotationData.stopValue}`}
+          config={plotConfig}
+          data={annotationData}
+          boundingReference={boundingReference}
+          width={width}
+          height={height}
+        />
+      ))}
+    </>
+  )
```

Comment:
Oops, yeah, sorry that was leftover from a previous iteration of the code. I had played around with different versions of this and forgot to clean this up.

Reply:
Turns out, we still need the fragment, because there is no parent element wrapping the array of elements. And that is the use case here -- no parent element.

Reply:
i can never remember the rules. I thought you were allowed to return a collection of elements even if it didn't have a parent